.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: Pulse CI
+on:
+  push:
+    branches-ignore:
+    - _**
+  pull_request:
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    container: mtzguido/dev-base
+    steps:
+      - run: echo "HOME=/home/user" >> $GITHUB_ENV
+      - uses: mtzguido/set-opam-env@master
+
+      - uses: actions/checkout@master
+        id: checkout-fstar
+        with:
+          path: FStar
+          repository: FStarLang/FStar
+          ref: master
+
+      - name: Try fetch built F*
+        id: cache-fstar
+        uses: actions/cache/restore@v4
+        with:
+          path: FStar
+          key: FStar-${{ runner.os }}-${{ runner.arch }}-${{ steps.checkout-fstar.outputs.commit }}
+
+      - name: Build F*
+        if: steps.cache-fstar.outputs.cache-hit != 'true'
+        run: |
+          make -C FStar ADMIT=1 -skj$(nproc) 1
+          make -C FStar ADMIT=1 -skj$(nproc) bootstrap
+
+      - name: Save built F*
+        if: steps.cache-fstar.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: FStar
+          key: FStar-${{ runner.os }}-${{ runner.arch }}-${{ steps.checkout-fstar.outputs.commit }}
+
+      - run: echo "FSTAR_HOME=$(pwd)/FStar" >> $GITHUB_ENV
+
+      - uses: actions/checkout@master
+        id: checkout-karamel
+        with:
+          path: karamel
+          repository: FStarLang/karamel
+          ref: master
+
+      - name: Try fetch built karamel
+        id: cache-karamel
+        uses: actions/cache/restore@v4
+        with:
+          path: karamel
+          key: karamel-${{ runner.os }}-${{ runner.arch }}-${{ steps.checkout-karamel.outputs.commit }}
```

Comment:
I wonder when we'll run into issues because the key doesn't capture all dependencies.  This is missing the ocaml and F* versions (Karamel builds checked files).

Reply:
Yeah, indeed there's many things possibly not captured here, I wonder what's a reasonable cutoff. A stopgap could be to add today's date here, so we at least don't inadvertently carry a stale cache too long.

Note however we are running in a fixed container, so the OCaml version (and libc and whatever) are fixed. (The F* version should indeed be included here though.) Unless we want to share this cache between more workflows..?